### ISSUE TITLE
Crash in com.Metal.CompletionQueueDispatch

### ISSUE THREAD

**Issue by ePirat** [KhronosGroup/MoltenVK#510]
I'm trying to use MoltenVK with [libplacebo](https://github.com/haasn/libplacebo) on macOS and running into issues when resizing the window causing random crashes in the Metal Completion Dispatch Queue:

```
Crashed Thread:        2  Dispatch queue: com.Metal.CompletionQueueDispatch

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument
abort() called
```

Full crashlog can be found [here](https://gist.github.com/ePirat/dae31d549c743e642d7af6cec6bfe504).

The full source code for the application can be found [here](https://github.com/ePirat/MoltenPlacebo).

**Comment by billhollings** [comment]
@ePirat 

Can you post, to the cloud somewhere, the `libplacebo.dylib` library you are using? That will allow us to build your app without having to install everything needed to build `libplacebo.dylib`.

**Comment by ePirat** [comment]
@billhollings Sure, I've updated my project to make that easier, download and extract [3rdParty.zip](https://0x0.st/z-Qu.zip) and put the 3rdParty folder in the MoltenPlacebo repo root, next to the `MoltenPlacebo` and  `MoltenPlacebo.xcodeproj`, the Xcode project should then find the libraries.

Hope that helps.

**Comment by ePirat** [comment]
It might be related to [this bug](https://github.com/n8vm/Pluto/issues/11)?

**Comment by ePirat** [comment]
@billhollings Any update on this?

**Comment by billhollings** [comment]
@ePirat Thanks for the prod. I'll look at it as soon as I can.

**Comment by billhollings** [comment]
@ePirat This is fixed in PR #652 . Please retest and close this issue if it fixes your issue.

**Comment by ePirat** [comment]
Yes, I can't reproduce anymore with latest master, so it seems #652 fixed it! Thanks a lot!
